runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesOverlay.java

```diff
@@ -191,21 +197,56 @@ private Ellipse2D drawEllipse(Graphics2D graphics, int x, int y)
 
 	private void drawSkillImage(Graphics2D graphics, XpGlobe xpGlobe, int x, int y)
 	{
-		BufferedImage skillImage = iconManager.getSkillImage(xpGlobe.getSkill());
-
-		if (skillImage == null)
+		BufferedImage toDraw;
+		if (xpGlobe.getSkillIcon() != null && xpGlobe.getGlobeSize() == config.xpOrbSize())
 		{
-			return;
+			toDraw = xpGlobe.getSkillIcon();
+		}
+		else
+		{
+			toDraw = generateScaledIcon(xpGlobe.getSkill());
+			if (toDraw == null)
+			{
+				return;
+			}
+			xpGlobe.setSkillIcon(toDraw);
+			xpGlobe.setGlobeSize(config.xpOrbSize());
```

Comment:
Couldn't this just be moved inside the else condition above?

Reply:
I'm a bit confused what you mean by this. It is part of the else condition, and is only set if the image is properly created.

Reply:
Oh my, I read that wrong because of the above comment. Sorry about that!

Reply:
nws :smile: 
Always good to double check things anyway